engine/src/main/scala/cromwell/engine/workflow/profiler/WorkflowProfiler.scala

```diff
@@ -0,0 +1,94 @@
+package cromwell.engine.workflow.profiler
+
+import akka.actor._
+import akka.event.LoggingReceive
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.core.WorkflowId
+import cromwell.engine.WorkflowState
+import cromwell.engine.workflow.WorkflowActor.{InitializingWorkflowState, WorkflowActorStateAndData}
+import cromwell.engine.workflow.lifecycle.WorkflowExecutionActor.WorkflowExecutionActorStateAndData
+import cromwell.engine.workflow.{CromwellFsmStateAndData, WorkflowMetadataKeys}
+import cromwell.services.MetadataServiceActor._
+import org.joda.time.DateTime
+
+import scala.concurrent.ExecutionContext
+
+object WorkflowProfiler {
+
+  case class WorkflowManagerMetadataMessage(workflowId: String, submissionTime: String)
+
+  implicit class EnhancedServiceRegistry(val serviceRegistryActor: ActorRef) extends StrictLogging {
+    def publishToMetadataService(message: MetadataEvent)(implicit actorSystem: ActorSystem, ec: ExecutionContext): Unit /*Future[MetadataServiceResponse]*/ = {
+      import cromwell.util.PromiseActor._
+      // Since this will always publish, we expect the callee to just provide an event,
+      // and we wrap it into a `PutMetadataAction` before sending it to the MetadataService
+      serviceRegistryActor.askNoTimeout(PutMetadataAction(message)).mapTo[MetadataServiceResponse].recover {
+        case reason: Throwable =>
+          // TODO: Do we still proceed with the execution? What do we do with the exception here?
+          logger.warn(s"Failed to push event $message in the MetadataService: ${reason.getMessage}", reason)
+          throw reason
+      }
+    }
+  }
+
+  def props(workflowId: WorkflowId, serviceRegistryActor: ActorRef): Props = {
+    Props(new WorkflowProfiler(workflowId, serviceRegistryActor))
+  }
+}
+
+case class WorkflowProfiler(workflowId: WorkflowId, serviceRegistryActor: ActorRef) extends Actor with ActorLogging {
+
+  import WorkflowProfiler._
+
+  private val tag = s"WorkflowProfiler-${workflowId.toString}"
+  implicit val actorSystem = context.system
+  implicit val ec = context.dispatcher
+
+  override def preStart(): Unit = {
+    context.system.eventStream.subscribe(self, classOf[CromwellFsmStateAndData])
+  }
+
+  override def receive = LoggingReceive {
+    case workflowManagerActorMessage: WorkflowManagerMetadataMessage => handleMessageFromWorkflowManagerActor(workflowManagerActorMessage)
```

Comment:
I might remove this message I think, and simply add the required info (which really is just the submission time in the constructor)